sdk/ml/azure-ai-ml/tests/dsl/unittests/test_dsl_pipeline_with_specific_nodes.py

```diff
@@ -978,6 +1564,436 @@ def pipeline(iris_data, sample_rate):
+                                                                          {'name': 'department',
+                                                                           'type': 'String',
+                                                                           'value': 'Engineering'}],
+                                              'type': 'database'},
+                                   'task': 'import_data',
+                                   'type': 'data_transfer'}},
+                'outputs': {'pipeline_output': {'job_output_type': 'mltable'}},
+                'properties': {},
+                'settings': {'_source': 'DSL'},
+                'tags': {}}
+        }
+
+    def test_pipeline_with_data_transfer_export_database_job(self):
+        connection_target_azuresql = 'azureml:my_azuresql_connection'
+        table_name = "merged_table"
+        cosmos_folder = Input(type=AssetTypes.URI_FILE, path="azureml://datastores/my_cosmos/paths/source_cosmos")
+        inputs = {"source": cosmos_folder}
+        sink = {'type': 'database', 'connection': connection_target_azuresql, 'table_name': table_name}
+        data_transfer_job = DataTransferExportJob(
+            inputs=inputs,
+            sink=sink,
+            task=DataTransferTaskType.EXPORT_DATA,
+        )
+        data_transfer_job_func = to_component(job=data_transfer_job)
+
+        @dsl.pipeline(experiment_name="test_pipeline_with_data_transfer_export_database_job")
+        def pipeline(table_name, connection_target_azuresql):
+            node1 = data_transfer_job_func(source=cosmos_folder)
+            node1.sink = sink
+
+            source_snowflake = Database(table_name=table_name, connection=connection_target_azuresql)
+            node2 = data_transfer_job_func(source=cosmos_folder)
+            node2.sink = source_snowflake
+
+        pipeline1 = pipeline(table_name, connection_target_azuresql)
+        pipeline_rest_obj = pipeline1._to_rest_object()
+        pipeline_job1 = pipeline_rest_obj.as_dict()
+
+        pipeline_regenerated_from_rest = PipelineJob._load_from_rest(pipeline_rest_obj)
+
+        # pipeline1_dict will dump to component dict according to schema, but
+        # pipeline_regenerated_from_rest will only keep component id after call _to_rest_object()
+        omit_fields = ["jobs.*.component"]
+        pipeline1_dict = omit_with_wildcard(pipeline1._to_dict(), *omit_fields)
+        pipeline_regenerated_from_rest_dict = omit_with_wildcard(pipeline_regenerated_from_rest._to_dict(), *omit_fields)
+
+        assert pipeline1_dict == pipeline_regenerated_from_rest_dict
+
+        omit_fields = [
+            "properties.jobs.*.componentId",
+        ]
+        pipeline_job1 = omit_with_wildcard(pipeline_job1, *omit_fields)
+        assert pipeline_job1 == {
+            'properties': {
+                'display_name': 'pipeline',
+                'experiment_name': 'test_pipeline_with_data_transfer_export_database_job',
```

Comment:
may remove experiment_name which is not critical for datatransfer test. So we can save some lines.
If you would like it to act something like a comment, you can keep it

Reply:
will remove